### ISSUE TITLE
Context Not Working for Hsnips

### ISSUE THREAD

**Issue by ColtWhitley** [sleepymalc/VSCode-LaTeX-Inkscape#17]
Hi I am trying to get this system up and working for my own note taking, but for some reason the code snippets work regardless of the context I am writing in. This means that I cannot take notes in English without my words becoming replaced by expressions that are quite useful for equations but not explanations. If anyone knows why the context is no longer filtering please let me know!

Here is the code I have tried to fix the context issues:
```
global
function math(context) {
    return context.scopes.some(s => s.startsWith("meta.math"));
}
endglobal
```
```
global
function math(context) {
    return  context.scopes.findLastIndex(s => s.startsWith("meta.math")) > context.scopes.findLastIndex(s => s.startsWith("comment") || s.startsWith("meta.text.normal.tex"));
}
endglobal
```
```
global
function math(context) {
    return context.scopes.some(s => s.startsWith("math"));
}
endglobal
```

**Comment by sleepymalc** [comment]
The current scope of the location of your cursor can be seen by inspecting `Developer: Inspect Editor Tokens and Scopes` (by `shift`+`cmd`+`p` and searching for this). Please provide more info about in what scenario the snippets are falsely triggered and what's the corresponding scope.

For example, you should see the following when you're in the math environment:
![image](https://user-images.githubusercontent.com/60427058/211909187-81b16862-6856-444c-9a51-70ce1dd514b1.png)

Also, it helps to inspect the extension log via `Developer: Open Extension Logs Folder`, to see whether there are any weird errors.

**Comment by ColtWhitley** [comment]
I coppie your new version of the hsnips file and it resolved itself, I must have accidentally put a typo somewhere

**Comment by ColtWhitley** [comment]
The snippets are working even when the context doesn't contain meta.math

<img width="445" alt="Screen Shot 2023-01-11 at 12 25 37 PM" src="https://user-images.githubusercontent.com/75457827/211911096-45c5788a-4904-4805-9fab-9fc75ea95c19.png">

In the extension Logs folder the only thing I see is this but I don't know what it means:
`2023-01-11 12:23:35.867 [error] TypeError: Cannot read properties of undefined (reading 'range')
    at HSnippetInstance.update (/Users/coltwhitley/.vscode/extensions/draivin.hsnips-0.2.8/src/hsnippetInstance.ts:213:19)`

**Comment by sleepymalc** [comment]
Yes, this is the only issue Hypersnips currently has and remains unresolved. See [here](https://github.com/draivin/hsnips/issues/79). A quick fix might be simply closing the current tab and reopening it again: this usually works, and I suggest you simply put one more tab (some random file or just `latex.hsnips` for a quick change), so whenever you encounter, when closing the tab, windows will not be re-adjust (might sound cryptic but you'll get the idea when you try).

**Comment by ColtWhitley** [comment]
Dam that sucks, but thank you! it was definitely an intermittent issue so I was really struggling to resolve :) Have a good day
